presto-kudu/src/test/java/io/prestosql/plugin/kudu/TestKuduIntegrationGroupedExecution.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kudu;
+
+import io.prestosql.Session;
+import io.prestosql.cost.StatsAndCosts;
+import io.prestosql.metadata.Metadata;
+import io.prestosql.sql.planner.Plan;
+import io.prestosql.sql.planner.plan.ExchangeNode;
+import io.prestosql.testing.AbstractTestQueryFramework;
+import io.prestosql.testing.DistributedQueryRunner;
+import io.prestosql.testing.QueryRunner;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.util.function.Consumer;
+
+import static io.prestosql.SystemSessionProperties.COLOCATED_JOIN;
+import static io.prestosql.SystemSessionProperties.CONCURRENT_LIFESPANS_PER_NODE;
+import static io.prestosql.SystemSessionProperties.DYNAMIC_SCHEDULE_FOR_GROUPED_EXECUTION;
+import static io.prestosql.SystemSessionProperties.ENABLE_DYNAMIC_FILTERING;
+import static io.prestosql.SystemSessionProperties.GROUPED_EXECUTION;
+import static io.prestosql.plugin.kudu.KuduQueryRunnerFactory.createKuduQueryRunner;
+import static io.prestosql.sql.planner.optimizations.PlanNodeSearcher.searchFrom;
+import static io.prestosql.sql.planner.planprinter.PlanPrinter.textLogicalPlan;
+import static java.lang.String.format;
+
+public class TestKuduIntegrationGroupedExecution
+        extends AbstractTestQueryFramework
+{
+    public static final String KUDU_GROUPED_EXECUTION = "kudu.grouped_execution";
+    private TestingKuduServer kuduServer;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        kuduServer = new TestingKuduServer();
+        return createKuduQueryRunner(kuduServer, "test_grouped_execution");
+    }
+
+    @AfterClass(alwaysRun = true)
+    public final void destroy()
+    {
+        kuduServer.close();
+    }
+
+    @Test
+    public void testGroupedExecutionJoin()
+    {
+        Session groupedExecutionSessionProperties = Session.builder(getSession())
```

Comment:
It would be the best to make it a default session in kudu query runner you are using here. I mean it would be best to pass this session to `createKuduQueryRunner`

Reply:
Agree!

Reply:
agree!